src/foxops/services/template.py

```diff
@@ -0,0 +1,13 @@
+from foxops.engine.models.template_config import TemplateConfig
+from foxops.hosters import Hoster
+
+
+class TemplateService:
+    def __init__(self, hoster: Hoster) -> None:
+        self.hoster = hoster
+
+    async def get_template_variables(self, template_repository: str, template_version: str) -> dict[str, str]:
+        async with self.hoster.cloned_repository(template_repository, refspec=template_version) as repo:
+            template_config = TemplateConfig.from_path(repo.directory / "fengine.yaml")
+
+        return {k: v.get("default", "") for k, v in template_config.model_dump().get("variables", {}).items()}
```

Comment:
why the casting into a dict? We can have better type safety in the code by just doing

```suggestion
        return {k: v.get("default", "") for k, v in template_config.variables.items()}
```

but then, also we have to be aware here that variables can be nested, complex objects. So this logic would have to be more recursive.

I would recommend adding functionality into the `*VariableDefinition` classes for getting an empty "example object" of that variable. This should then be easy to combine "up the tree" to ultimately get an "empty example" of a full template config.

Reply:
just added a commit to this branch which added a `mock_data()` method on the TemplateConfig object - including a test for it :-) check it out!